### PR TITLE
Handle TCP data transmission

### PR DESCRIPTION
**Pull request by laochanlam**
Simply handled the TCP data packet then replied ACK packet while ignoring data parsing.

### CODE REVIEW

File: src/tcp.c [hunk 1]
```diff
@@ -241,7 +241,17 @@ static int tcp_fsm(struct tcp_conn_tcb *conn, struct tcp_hdr *rs)
...
-
+        if (rs->tcp_flags & TCP_ACK && rs->tcp_flags & TCP_PSH &&
+            rs->tcp_seqno == conn->recv_next && rs->tcp_ack_num == conn->send_next) { /* data handling */
+            rs->tcp_flags &= ~TCP_PSH;
+            rs->tcp_ack_num = rs->tcp_seqno + (bsize - tcp_hdr_size(rs));
+            rs->tcp_seqno = conn->send_next;
+            
+            conn->recv_next = rs->tcp_ack_num;
+            conn->send_next = rs->tcp_seqno; 
+            /* TODO upload payload to application layer for further pasing */
```
Comment (jserv) [inline src/tcp.c]: The comment is misleading. You can say, "forward the payload to application layer" instead.

File: src/tcp.c [hunk 2]
```diff
@@ -241,7 +241,17 @@ static int tcp_fsm(struct tcp_conn_tcb *conn, struct tcp_hdr *rs)
         return tcp_hdr_size(rs);
     case TCP_ESTABLISHED:
         LOG(LOG_INFO, "TCP state: TCP_ESTABLISHED");
-
+        if (rs->tcp_flags & TCP_ACK && rs->tcp_flags & TCP_PSH &&
```
Comment (jserv) [inline src/tcp.c]: Use parentheses `(` `)` to emphasize each condition in this statement.

File: src/tcp.c (hunk 1 above)
Comment (laochanlam) [reply src/tcp.c]: Ok!

**Comment by laochanlam** [comment]
I also wrote a TCP socket program for test. 
Should I add the script to the directory ```tests/``` on this pull request?

File: src/tcp.c (hunk 2 above)
Comment (laochanlam) [reply src/tcp.c]: Ok, will fix it in the later commit.

**Comment by jserv** [comment]
If you have TCP test program(s), just append to directory `tests`.

**Pull request merged**